### ISSUE TITLE
Platform push don't throw errors during process

### ISSUE THREAD

**Issue by byvlad** [platformsh/platformsh-cli#1065]
Hi!
The main problem that when you call platform push, and during the process something fails, this script doesn't throw error.
As you can see on the screenshots, usually, scripts throw errors.

![1](https://user-images.githubusercontent.com/935809/123785408-29797900-d8e1-11eb-980f-3f4697e50ad5.jpg)
![2](https://user-images.githubusercontent.com/935809/123785415-2bdbd300-d8e1-11eb-884c-66551a897121.jpg)

**Comment by pjcdawkins** [comment]
Yes. The issue is that the "push" succeeded even though the subsequent deployment didn't. Currently the platform push command passes through the exit status from git push. Perhaps the git API would need to implement some kind of feature to report the deployment failure via the git exit status.
